presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -616,6 +618,26 @@ public PlanWithProperties visitFilter(FilterNode node, PreferredProperties prefe
             return rebaseAndDeriveProperties(node, planChild(node, preferredProperties));
         }
 
+        @Override
+        public PlanWithProperties visitSequence(SequenceNode node, PreferredProperties preferredProperties)
+        {
+            List<PlanWithProperties> leftPlans = node.getCteProducers().stream()
+                    .map(source -> accept(source, PreferredProperties.any()))
+                    .collect(ImmutableList.toImmutableList());
+            PlanWithProperties rightPlan = accept(node.getPrimarySource(), PreferredProperties.any());
```

Comment:
Why right plan use `PreferredProperties.any()`, shouldn't it conform to what we are supposed to use for the primary source node?

Reply:
Yes I think this makes sense